File: nav/src/main/java/com/example/nav/test/MainActivity5.java
package com.example.nav.test;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.adapters.TextViewBindingAdapter;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.example.nav.R;

public class MainActivity5 extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityQueueManager.getInstance().pushActivity(this);
        setContentView(R.layout.activity_main5);

        findViewById(R.id.tv).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity5.this, MainActivity6.class);
                startActivity(intent);
            }
        });

        Log.e("MainActivity-5", ActivityQueueManager.getInstance().getAll().toString());
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        ActivityQueueManager.getInstance().popActivity(this);
    }
}

File: nav/src/main/java/com/example/nav/MainActivity5.kt
package com.example.nav

import android.app.ProgressDialog.show
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.coroutines.*
import kotlin.coroutines.CoroutineContext

class MainActivity5 : AppCompatActivity() {


    private val mainScope= MainScope()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main9)


//        runBlocking {  }
//        withContext()
//        withTimeout()
//        launch
//        async
//        Dispatchers  调度
//        CoroutineScope  作用域 它会创建一个协程作用域并且在所有已启动子协程执行完毕之前不会结束。
//        CoroutineContext  上下文
//        suspend  挂起函数  某些时刻挂起和恢复


    }







    override fun onDestroy() {
        super.onDestroy()
        mainScope.cancel()
    }


}